tests/cases/compiler/narrowingTypeofUndefined.ts

```diff
@@ -0,0 +1,5 @@
+declare const a: { error: { a: string }, result: undefined } | { error: undefined, result: { b: number } }
+
+if (typeof a.error === 'undefined') {
+    a.result.b; // ok
+}
```

Comment:
Can you add

1. an `else` branch demonstrating that you can access `a.error.string`, and
2. another `if`/`else` statement but with a `!==`?